### PR TITLE
add subclasspicker attribute and drawer

### PR DESCRIPTION
**Pull request by ThibNach**
Add the SubClassPicker Atribute.
Use it with the SerializeRerefence Attribute to chose a non-monobehaviour class 

image

- - -
This change is [Reviewable](https://reviewable.io/reviews/FishingCactus/UnityCommonCode/104)

### CODE REVIEW

**Review by Skeloson — APPROVED** [review]

**Comment by LeanderSimonart** [comment]
can you explain a scenario where this is used for please?

**Review by TheEmidee — APPROVED** [review]
:lgtm:

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/FishingCactus/UnityCommonCode/104#-:-Nj6nP6eDHAlBFde8QhL:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @LeanderSimonart and @MaxDaniels)

File: Editor/AttributesDrawer/SubClassPickerAttributeDrawer.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+            public override void OnGUI(
+                Rect position,
+                SerializedProperty property,
+                GUIContent label
+                )
+            {
+                Type field_type = fieldInfo.FieldType;
+
+                string type_name = property.managedReferenceValue?.GetType().Name ?? DefaultDisplay;
```
Comment (lucas-baran) [inline Editor/AttributesDrawer/SubClassPickerAttributeDrawer.cs:48]: Isn't `.Name` throwing a null ref if `managedReferenceValue` is null?

**Comment by ThibNach** [comment]
> can you explain a scenario where this is used for please?
@LeanderSimonart 
For badminton we have 5 minigames, we load the minigame with a scriptable object that contains the common parameters to all the minigames. One of those parameters is the specific minigame controller itself ( MiniGameControllerBase in the screenshot ).
MiniGameControllerBase is not a MonoBehaviour so with this attribute I can pick the controller I will need to instanciate for my minigame.
For MiniGame_Snake => SnakeController, for MiniGame_Duck => DuckController, for MiniGame_Balloons => BalloonsController, etc....

**Review by lucas-baran — APPROVED** [review]

**Comment by ThibNach** [comment]
*[`Editor/AttributesDrawer/SubClassPickerAttributeDrawer.cs` line 48 at r1](https://reviewable.io/reviews/FishingCactus/UnityCommonCode/104#gh-1390771227:-Nj6wNVwEzjf7hslZlQe:b-rulf7f) ([raw file](https://github.com/FishingCactus/UnityCommonCode/blob/fbf2283a83a0079ec790d305bbdd6d68eeea94a0/Editor/AttributesDrawer/SubClassPickerAttributeDrawer.cs#L48)):*
<details><summary><i>Previously, lucas-baran (Lucas Baran) wrote…</i></summary><blockquote>

Isn't `.Name` throwing a null ref if `managedReferenceValue` is null?
</blockquote></details>

nope

**Review by rascar-capac — APPROVED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/FishingCactus/UnityCommonCode/104)* status: all files reviewed, 1 unresolved discussion (waiting on @fishingcactus-nathan-istace, @LeanderSimonart, @lucas-baran, @MaxDaniels, @PetitT, and @Skeloson)

**Review by MaxDaniels — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/FishingCactus/UnityCommonCode/104)* status: all files reviewed, 1 unresolved discussion (waiting on @fishingcactus-nathan-istace, @LeanderSimonart, @lucas-baran, @PetitT, and @Skeloson)

**Review by Skeloson — APPROVED** [review]

**Review by LeanderSimonart — APPROVED** [review]

**Pull request merged**